src/display/canvas.js

```diff
@@ -1443,17 +1448,21 @@ var CanvasGraphics = (function CanvasGraphicsClosure() {
 
       var simpleFillText =
         current.textRenderingMode === TextRenderingMode.FILL &&
-        !font.disableFontFace;
+        !font.disableFontFace && !current.patternFill;
 
       ctx.save();
-      ctx.transform.apply(ctx, current.textMatrix);
-      ctx.translate(current.x, current.y + current.textRise);
-
+      let patternTransform;
       if (current.patternFill) {
-        // TODO: Some shading patterns are not applied correctly to text,
-        //       e.g. issues 3988 and 5432, and ShowText-ShadingPattern.pdf.
-        ctx.fillStyle = current.fillColor.getPattern(ctx, this);
+        // TODO: Patterns are not applied correctly to text if a system font
```

Comment:
Nit: Let's replace `system font` with `non-embedded font` instead here, since that's more inline with the notation used elsewhere in PDF.js.